### DISCUSSION TITLE
Filter out all records for a specific user

### DISCUSSION THREAD

**Discussion by lavvy** [pocketbase/pocketbase#5202] [Q&A]
I am using PocketBase v0.22.16 and dart sdk v0.18.1. So far all features work okay.
But trying to get all records associated to a particular user from a collection of records seems not responding.
It doesn't filter, instead pulls all records for all users. 
I don't know whether am wrong or its a bug.

`
final records = await pb.collection('configuration').getFullList(
  //filter: pb.filter("userid = {:id}", {"id": pb.authStore.model.id})
      filter: "userid = '${pb.authStore.model.id}'");
`

![api_rules](https://github.com/pocketbase/pocketbase/assets/7013091/6c4e5950-9e4f-4b2d-bb07-e542882ac1d9)
![db_gui](https://github.com/pocketbase/pocketbase/assets/7013091/8bc89abf-5e8e-40c0-91d9-789589cc2daf)
![Screenshot_10-7-2024_125422_flutlab io](https://github.com/pocketbase/pocketbase/assets/7013091/bbe6ddf1-a339-4ce6-a980-f9e19e1cc302)

[pb_schema.json](https://github.com/user-attachments/files/16165547/pb_schema.json)

**Comment by ganigeorgiev** [comment]
It works correctly for me, I'm not able to reproduce it.

Make sure that you are sending the request to the correct PocketBase instance.

Here is the minimal Dart example that I've tried (aka. `dart example.dart`):

```dart
// example.dart
import "package:pocketbase/pocketbase.dart";

void main() {
  final pb = PocketBase("http://127.0.0.1:8090");

  pb.collection("configuration").getFullList(filter: "userid = 'abc'")
    .then((result) {
      // it correctly returns only the records matching the filter
      print("Result: $result");
    }).catchError((dynamic error) {
      print("Error: $error");
    });
}
```

The easiest way to rule out PocketBase filter issue is to try the filter directly in the Admin UI searchbar.

If you have custom hooks, make sure that they don't intercept the default request response.

You can also try to start the executable with `--dev` and inspect the generated SQL queries.

**Reply by ganigeorgiev** [reply]
Also if you have a reverse proxy (nginx, apache, caddy, etc.) make sure that the query parameters are properly proxied back to the PocketBase server.

**Comment by lavvy** [comment]
""" 
The easiest way to rule out PocketBase filter issue is to try the filter directly in the Admin UI searchbar. 
""""
Could you give me a string representation of that command let me paste in the search bar.

Am certain its goin to the right pb instance . Every other data comes back correctly

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand. The searchbar accept the same filter string as the one you pass in the SDK (obviously in the Admin UI you don't have `pb.authStore.model.id` and you have to replace it with your string id).

**Reply by lavvy** [reply]
![Screenshot_10-7-2024_132357_asb1 fly dev](https://github.com/pocketbase/pocketbase/assets/7013091/abbec72d-5a4c-4df8-bff2-cd45eabb34db)
![Screenshot_10-7-2024_132253_asb1 fly dev](https://github.com/pocketbase/pocketbase/assets/7013091/8a5fcaed-c8fc-463d-820f-c898406f66ed)

**Reply by ganigeorgiev** [reply]
There is definitely something wrong with your environment. Again, double check your setup if you have a proxy or custom hooks but I don't think I can help you further.

Here is a working example from the live demo (test@example.com/123456): 
https://pocketbase.io/_/#/collections?collectionId=XTpascjA7jyzB88&filter=message%3D%22pong%22&sort=-created
![filter_screenshot](https://i.imgur.com/wT1G3nd.png)

**Reply by lavvy** [reply]
Ok . I noticed the search bar has autofill which mine hasn't. 
I may have to upgrade to 0.22.17 and check again.

Secondly thanks for the demo site, but i tried pointing my app to the server  for a test but didnt connect.
Is the a different port for connections

**Reply by ganigeorgiev** [reply]
There are no changes to the filter or the UI autocomplete between v0.22.16 and v0.22.17 (_just tested it with v0.22.16 and it works correctly for me_).

> Secondly thanks for the demo site, but i tried pointing my app to the server for a test but didnt connect.

There is no need to specify port (for https by default is 443), or in other words:

```dart
import 'package:pocketbase/pocketbase.dart';

final pb = PocketBase('https://pocketbase.io');
```

I'll stop engage to avoid further confusion, but again - if you have a proxy (nginx, apache, caddy, etc.) or custom hooks double check their configuration.

**Reply by lavvy** [reply]
okay
